src/libsync/syncengine.cpp

```diff
@@ -1642,6 +1642,18 @@ void SyncEngine::setLocalDiscoveryOptions(LocalDiscoveryStyle style, std::set<QB
 {
     _localDiscoveryStyle = style;
     _localDiscoveryPaths = std::move(paths);
+
+    // Normalize to make sure that no path is a prefix of another
+    QByteArray prev;
+    auto it = _localDiscoveryPaths.begin();
+    while(it != _localDiscoveryPaths.end()) {
+        if (!prev.isNull() && it->startsWith(prev) && (prev.endsWith('/') || *it == prev || it->at(prev.size()) == '/')) {
```

Comment:
I don't think this is right because of the dreaded ' ' < '/'. So: `"foo" < "foo bar" < "foo/bar"` and iterating would not remove the redundant "foo". The way we solved this in selective sync was to ensure directories always have a trailing '/' I think.